aiidalab_widgets_base/structures.py

```diff
@@ -1,88 +1,138 @@
-
-        children = [self.file_upload, self.viewer, store]
-
+            get_structure = self.file_upload
+
+        if storable:
+            if node_class is None:
+                store = [self.btn_store, self.data_format, self.structure_description]
+            elif node_class not in self.DATA_FORMATS:
+                raise ValueError("Unknown data format '{}'. Options: {}".format(
+                    node_class, self.DATA_FORMATS))
+            else:
+                self.data_format.value = node_class
+                store = [self.btn_store, self.structure_description]
+        else:
+            store = [self.structure_description]
+        store = ipw.HBox(store)
+        children = [get_structure, supported_formats, self.viewer, store]
         super(StructureUploadWidget, self).__init__(
             children=children, **kwargs)
 
         self.file_upload.observe(self._on_file_upload, names='data')
+        self.select_example.observe(self._on_select_example, names=['value'])
         self.btn_store.on_click(self._on_click_store)
 
+    @staticmethod
+    def get_example_structures(examples):
+        if examples:
+            to_return=[("Select structure", False)]
+            to_return += [(s.split('/')[-1], s) for s in examples]
+            return to_return
+        else:
+            return []
+
     # pylint: disable=unused-argument
     def _on_file_upload(self, change):
-        self.tmp_folder = tempfile.mkdtemp()
-        tmp = self.tmp_folder + '/' + self.file_upload.filename
-        with open(tmp, 'w') as f:
+        self.file_path = tempfile.mkdtemp() + '/' + self.file_upload.filename
+        with open(self.file_path, 'w') as f:
             f.write(self.file_upload.data)
-        self.select_structure(name=self.file_upload.filename)
+        structure_ase = self.get_ase(self.file_path)
+        self.select_structure(structure_ase=structure_ase, name=self.file_upload.filename)
 
-    def select_structure(self, name):
-        structure_ase = self.get_ase(self.tmp_folder + '/' + name)
-        self.btn_store.disabled = False
-        if structure_ase is None:
+
+    def _on_select_example(self, change):
+        if self.select_example.value:
+            structure_ase = self.get_ase(self.select_example.value)
+            self.file_path = self.select_example.value
+        else:
+            structure_ase = False
+        self.select_structure(structure_ase=structure_ase, name=self.select_example.label)
+
+    def select_structure(self, structure_ase, name):
+        """ Select structure
+
+        :param structure_ase: ASE object containing structure
+        :type structure_ase: ASE Atoms
+        :param name: File name with extension but without path
+        :type name: str
+        """
+        self.name = name
+        self._structure_node = None
+        if not structure_ase:
             self.structure_ase = None
             self.btn_store.disabled = True
+            self.has_structure = False
```

Comment:
what about the `self.structure_description.value`, does it also need to be reset here?

I'm getting a bit uncomfortable with the fact that there are so many things to reset... we should think about whether we can reduce them.
Anyhow, I would suggest to split the resetting part of `select_structure` up into a `reset_structure` function that does nothing but reset the widget.

Reply:
Ok, I will add a reset function

Reply:
added `reset_structure` function and also put `self.structure_description.value = ''` inside it.